app/controllers/purchases_controller.rb

```diff
@@ -0,0 +1,54 @@
+require 'purchase/presenters/new_purchase_presenter'
+
+class PurchasesController < ApplicationController
+
+  def new
+    item = find_item(item_id)
+    purchase = new_purchase(item_id: item_id)
+    @presenter = create_presenter(purchase, item)
+  end
+
+  def create
+    purchase = new_purchase(email: buyer_email, item_id: item_id)
+    item = find_item(item_id)
+    @presenter = create_presenter(purchase, item)
+
+    if purchase.save
+      item.update_attributes(available: false)
+      send_email(purchase, item)
+      render :show
+    else
+      redirect_to "/items/#{item_id}/purchases/new"
```

Comment:
This should render "new", because you probably will want to show the user what went wrong once you fix your race condition that you have here.